### PR TITLE
Add local file persister

### PR DESCRIPTION
**Pull request by ankur22**
## What?

This adds a local file persister which will save files to the local disk. It implements the interface defined in https://github.com/grafana/xk6-browser/issues/1155.

## Why?

So that we can replace how we're currently writing screenshots to disk. This will help hide the details of where and how the files are saved, allowing us to easily work with different file persisters.

## Checklist

- [x] I have performed a self-review of my code
- [x] I have added tests for my changes
- [x] I have commented on my code, particularly in hard-to-understand areas

## Related PR(s)/Issue(s)

Updates: https://github.com/grafana/xk6-browser/issues/1156

### CODE REVIEW

File: storage/file_persister.go [hunk 1]
```diff
@@ -12,3 +15,49 @@ import (
...
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return fmt.Errorf("reading from the reader: %w", err)
+		}
+	}
+
+	return nil
+}
```
Comment (inancgumus) [inline storage/file_persister.go:63]: Why don't we just use `io.Copy` here?

File: storage/file_persister.go [hunk 2]
```diff
@@ -12,3 +15,49 @@ import (
...
+		return fmt.Errorf("creating a local directory %q: %w", dir, err)
+	}
+
+	f, err := os.OpenFile(cp, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0o600)
+	if err != nil {
+		return fmt.Errorf("creating a local file %q: %w", cp, err)
+	}
+	defer func() {
+		if err := f.Close(); err != nil {
+			logger.Errorf("LocalFilePersister:Persist", "closing the local file %q: %v", cp, err)
```
Comment (inancgumus) [inline storage/file_persister.go:38]: I think we can handle this also from the call site if we return an error here. So we don't have to pass a logger here. The call site can log the error if it sees it's a file close error. WDYT?

File: storage/file_persister.go (hunk 1 above)
Comment (ankur22) [reply storage/file_persister.go]: I have no idea how i missed that 😆. Good suggestion, added to be155a55e95efa05b7195b81d99c2cd717d54c91

File: storage/file_persister.go (hunk 2 above)
Comment (ankur22) [reply storage/file_persister.go]: Yeah, that's a nice idea! Added it to cfa2726d577b1a26a0233ba5a10dd540ded70ffe

In which case the `logger` can be removed 🎉

**Comment by ankur22** [comment]
Closing this and moving this to https://github.com/grafana/xk6-browser/pull/1197

**Pull request closed without merging**